src/ProcessedCodeCoverageData.php

```diff
@@ -69,9 +69,7 @@ public function markCodeAsExecutedByTestCase(string $testCaseId, RawCodeCoverage
         foreach ($executedCode->lineCoverage() as $file => $lines) {
             foreach ($lines as $k => $v) {
                 if ($v === Driver::LINE_EXECUTED) {
-                    if (empty($this->lineCoverage[$file][$k]) || !in_array($testCaseId, $this->lineCoverage[$file][$k], true)) {
-                        $this->lineCoverage[$file][$k][] = $testCaseId;
-                    }
+                    $this->lineCoverage[$file][$k][] = $testCaseId;
```

Comment:
If this is safe, then the checks on 83/91 (pre-patch) 81/89 (post-patch) I think should be safe to remove as well, the principle behind adding those was cargo-culting over from this one

Reply:
Not sure what happens in this case but it's possible for the same test to be added to this array twice now.

If the array was keyed by the test id, you could set the key to be the id and the value to be true ? Then use array_keys before processing. 

Either way that's a huge improvement